torch/_tensor_str.py

```diff
@@ -297,15 +297,24 @@ def get_summarized_data(self):
     else:
         return torch.stack([get_summarized_data(x) for x in self])
 
-def _str_intern(inp):
-    self, tangent = torch.autograd.forward_ad.unpack_dual(inp)
-    prefix = "nested_tensor(" if self.is_nested else 'tensor('
+def _str_intern(inp, *, tensor_contents=None):
```

Comment:
So passing the `tensor_contents` completely replace what the usual impl puts inside the () ?

Reply:
It replaces the portion inside the () that shows what the tensor contains (i.e. its values). Any extra flags will still show up in the string repr:
```
MyTensor(<tensor_contents>, device='cuda:0', dtype=torch.float64, layout=torch.channels_last)
```

I'm happy to rename `tensor_contents` to be clearer what part it replaces or take a different approach to string repr customization if needed.